# İP6 — Tutarsızlık Analizi ve Karşılaştırmalı Validasyon

**Görev (Bölüm 10, Görev 8):** Tutarsızlık Analizi ve Karşılaştırmalı Validasyon
**Sorumlu:** Samet Koca | **Tarih:** 18 Temmuz 2026

**Girdi:** `data/processed/master_akts_final.xlsx` (İP5 çıktısı) + `data/processed/master_bloom.xlsx`

Bu notebook, metodolojideki Aşama 2 (Tutarsızlık Analizi) ve Aşama 4 (Karşılaştırma/Validasyon) adımlarını ve H1/H2/H4 hipotez testlerini çalıştırır. **H3, öğrenci iş yükü anketi (Kanal C) toplanmadığı için test edilememektedir** — bkz. rapor Sınırlılıklar bölümü.

In [1]:
import sys
from pathlib import Path

BASE = Path.cwd().resolve().parents[1]
sys.path.insert(0, str(BASE / "ip6-validasyon" / "src"))

import ip6_validasyon as ip6

ip6.main()

İP6 — TUTARSIZLIK ANALİZİ VE KARŞILAŞTIRMALI VALİDASYON



Yüklendi: 4133 ders (master_akts_final.xlsx + master_bloom.xlsx birleşimi)

------------------------------------------------------------
AŞAMA 2(a) / H2 — Kurum-İçi Tutarsızlık (aynı isimli ders, farklı bölüm)
------------------------------------------------------------
  Aynı isimle birden fazla bölümde geçen ders sayısı: 244
  Bunlardan ±2 AKTS aralığını AŞANLAR: 28 (11.5%)
  Ortalama AKTS aralığı (bu gruplarda): 1.21

------------------------------------------------------------
AŞAMA 2(b) — Etkinlik Bazlı Etki
------------------------------------------------------------
  has_lab        var=+19.96%  yok=+8.98%  fark=+10.98%  p=0.0000
  has_proje      var=+21.85%  yok=+8.36%  fark=+13.49%  p=0.0000
  has_uygulama   var=+16.57%  yok=+8.61%  fark=+7.95%  p=0.0002
  has_atolye     var=+44.18%  yok=+9.18%  fark=+35.00%  p=0.0408
  has_odev       var=+12.83%  yok=+8.61%  fark=+4.22%  p=0.0024
  has_sinav      var=+9.50%  yok=+3.02%  fark=+6.47%  p=0.0050
  has_sunum      var=+15.83%  yok

### Çıktı Dosyalarını İncele

In [2]:
import pandas as pd

pd.read_excel(BASE / 'ip6-validasyon' / 'ayni_isimli_ders_tutarsizligi.xlsx').head(10)

,DersAdi,n_bolum,n_kayit,akts_min,akts_max,akts_std,akts_araligi
0,Bitirme Çalışması,17,17,1,16,3.314407,15
1,Spor Bilimlerinde Araştırma Projesi I,2,2,4,10,4.242641,6
2,Bilgisayar Grafikleri,2,2,2,6,2.828427,4
3,Bitirme Projesi,3,3,2,6,2.081666,4
4,Osmanlı Türkçesi I,2,2,2,6,2.828427,4
5,Osmanlı Türkçesi II,2,2,2,6,2.828427,4
6,Mesleki İngilizce I,7,7,2,6,1.676163,4
7,Mesleki İngilizce II,7,7,2,6,1.345185,4
8,Yazarlık I,2,2,3,7,2.828427,4
9,Mesleki İngilizce III,2,2,2,6,2.828427,4


In [3]:
pd.read_excel(BASE / 'ip6-validasyon' / 'etkinlik_bazli_etki.xlsx')

,etkinlik,n_var,ort_sapma_pct_var,n_yok,ort_sapma_pct_yok,fark,mannwhitney_p,anlamli_mi(p<0.05)
0,has_atolye,20,44.18,4113,9.18,35.00,0.040815,True
1,has_proje,303,21.85,3830,8.36,13.49,0.000000,True
2,has_lab,139,19.96,3994,8.98,10.98,0.000010,True
3,has_uygulama,382,16.57,3751,8.61,7.95,0.000170,True
4,has_sunum,312,15.83,3821,8.82,7.01,0.017526,True
5,has_sinav,4039,9.50,94,3.02,6.47,0.005050,True
6,has_odev,723,12.83,3410,8.61,4.22,0.002360,True


In [4]:
pd.read_excel(BASE / 'ip6-validasyon' / 'inceleme_kuyrugu_ilk50.xlsx').head(10)

,Katalogid,DersAdi,Fakulte,AKTSKredi,adaptif_akts,sapma_yuzde,gerekce
0,9509110,Sosyal Sorumluluk Uygulamaları,Eğitim,3,24.5566,718.55,AKTS_mevcut=3.0 | rf_tahmin=33.31 | kural_akts...
1,9509084,Bilişim Teknolojileri,Eğitim,4,24.3021,507.55,AKTS_mevcut=4.0 | rf_tahmin=31.84 | kural_akts...
2,715187,AHŞAP ATÖLYESİ 2,Güzel Sanatlar,4,22.2383,455.96,AKTS_mevcut=4.0 | rf_tahmin=29.77 | kural_akts...
3,1701232,Araştırma Projesi,Spor Bilimleri,3,11.3965,279.88,AKTS_mevcut=3.0 | rf_tahmin=14.45 | kural_akts...
4,207536,Sayısal Tasarım Laboratuvarı,Mühendislik,1,3.7055,270.55,AKTS_mevcut=1.0 | rf_tahmin=4.68 | kural_akts=...
5,711070,Bitirme Çalışması,Güzel Sanatlar,2,7.3389,266.94,AKTS_mevcut=2.0 | rf_tahmin=9.26 | kural_akts=...
6,731005,Teknik Çizim Perspektif I,Güzel Sanatlar,1,2.7842,178.42,AKTS_mevcut=1.0 | rf_tahmin=3.55 | kural_akts=...
7,2105340,PSİKOLOJİ,Spor Bilimleri,3,7.9860,166.20,AKTS_mevcut=3.0 | rf_tahmin=9.67 | kural_akts=...
8,9507003,Teknik Çizim ve Perspektif-I,Güzel Sanatlar,2,5.3183,165.92,AKTS_mevcut=2.0 | rf_tahmin=6.71 | kural_akts=...
9,721127,Tipografi Yazı III,Güzel Sanatlar,4,10.4294,160.74,AKTS_mevcut=4.0 | rf_tahmin=12.69 | kural_akts...


### Bulgular

- **H1 DESTEKLENİYOR:** Lab/proje içeren derslerin ortalama sapması (+19.9%), diğer derslerden (+8.2%) anlamlı derecede yüksek (Mann-Whitney p<0.001).
- **H4 DESTEKLENİYOR:** Bloom düzeyi ile sapma arasında pozitif, anlamlı bir ilişki var (Spearman rho=0.16, p<0.001; Kruskal-Wallis p<0.001).
- **H2 (proxy):** Kurum-içi aynı isimli derslerin %11.5'i ±2 AKTS aralığını aşıyor — kurumlar arası (Kanal B) veri toplanınca yeniden test edilmeli.
- **H3: test edilemedi** (ground truth/anket verisi yok).

Tam rapor: `ip6-validasyon/IP6_SONUC_RAPORU.md`.